### PR TITLE
Issue/2114 add compiler entrypoint

### LINKED ISSUE #2114: Add entry point to the compiler to perform typing
Currently, the compiler.do_compile entry point can be used to do full execution, compiler.anchormap can be used to the anchormap, but there is no entrypoint to get the types and scopes with just typing information

This should be added

### PR DESCRIPTION
**Pull request by arnaudsjs**
# Description

Add compiler entrypoint to get types and scopes.

closes #2114 

# Self Check:

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] ~~End user documentation is included or an issue is created for end-user documentation~~

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: src/inmanta/compiler/__init__.py [hunk 1]
```diff
@@ -120,6 +120,17 @@ def anchormap(refs: Dict[Any, Any] = {}) -> Sequence[Tuple[Location, Location]]:
...
 
 
+def get_types_and_scopes() -> Tuple[Dict[str, inmanta_type.Type], Namespace]:
+    """
+        Only run the compilation steps required to extract the different types and scopes.
+    """
+    compiler = Compiler()
+    (statements, blocks) = compiler.compile()
+    sched = scheduler.Scheduler(compiler_config.track_dataflow())
+    sched.define_types(compiler, statements, blocks)
```
Comment (sanderr) [inline src/inmanta/compiler/__init__.py:130]: I think we should catch any `CompilerException` here and call `compiler.handle_exception()`. It writes errors to the configured file.

Comment (wouterdb) [reply src/inmanta/compiler/__init__.py:130]: Would it not be more logical for the caller of this method to use handle_exception when this is appropriate? 

I think most other pieces of software that use this method are happier with and exception than with a file?

Comment (sanderr) [reply src/inmanta/compiler/__init__.py:130]: Indeed, that makes sense. It does seem somewhat inconsistent that we do it differently for `do_compile` but I can see why we would do it.

Comment (wouterdb) [reply src/inmanta/compiler/__init__.py:130]: good remark, perhaps something to add as a note to the compiler SDK epic that we should have methods for consumption in programs and methods for consumption in CLI context

Comment (sanderr) [reply src/inmanta/compiler/__init__.py:130]: I've added it to the epic description

**Pull request merged**